### PR TITLE
Fix/tfn 423 persist description value

### PR DESCRIPTION
**Pull request by gilesdrinkall**
# Description

-  For both the Single Ticket journey and the Period Ticket (Multiple Services) journey, the purpose of this ticket is to ensure that service description values (as a result of calls to the TNDS Service table) are included in the JSON that is written up to S3. This will mean that the NeTEX generator functions do not need to make database calls for this data. Changes to the Single Ticket generator code were required as a result of the service description now coming from the Matching JSON data (as opposed to from a call to the database). 

# Testing instructions

-   I ran the site locally via the 'fdbt-dev' repo and checked that the JSON files written to S3 contained values for service description as expected.

# Type of change

-   [X] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

# Checklist:

-   [X] Able to run pr locally
-   [ ] Lighthouse performed
-   [ ] Followed acceptance criteria
-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my own code
-   [ ] I have made corresponding changes to the documentation if applicable
-   [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/pages/api/singleOperator.ts [hunk 1]
```diff
@@ -50,10 +50,14 @@ export default (req: NextApiRequest, res: NextApiResponse): void => {
...
+
         const requestBody: { [key: string]: string } = req.body;
 
         Object.entries(requestBody).forEach(entry => {
-            const data: ServicesInfo = { lineName: entry[0], startDate: entry[1] };
+            const checkedBoxValues = entry[1].split('/');
+            const serviceDescription = checkedBoxValues[0];
+            const startDate = checkedBoxValues[1];
+            const data: ServicesInfo = { lineName: entry[0], startDate, serviceDescription };
             checkedServiceList.push(data);
```
Comment (dannydaviesIW) [inline src/pages/api/singleOperator.ts:61]: Could we have the startDate and serviceDescription declared inline?

File: src/pages/matching.tsx [hunk 2]
```diff
@@ -20,6 +20,7 @@ export interface BasicService {
     lineName: string;
     nocCode: string;
     operatorShortName: string;
+    description: string;
```
Comment (dannydaviesIW) [inline src/pages/matching.tsx:23]: Just having a poke around and it looks like the matching.ts API has an interface which might also need changing

File: src/pages/api/singleOperator.ts (hunk 1 above)
Comment (gilesdrinkall) [reply src/pages/api/singleOperator.ts:63]: Declared in line now.

File: src/pages/matching.tsx [hunk 3]
```diff
@@ -150,6 +151,7 @@ export const getServerSideProps = async (ctx: NextPageContext): Promise<{ props:
                 lineName,
                 nocCode,
                 operatorShortName: service.operatorShortName,
+                description: service.serviceDescription,
```
[the remark this answers is not in the document]
Comment (gilesdrinkall) [reply src/pages/matching.tsx]: Changed key to 'serviceDescription'.

File: src/pages/matching.tsx (hunk 2 above)
Comment (gilesdrinkall) [reply src/pages/matching.tsx]: 'serviceDescription now added as a key to MatchingData.

**Pull request merged**